src/main/java/gregtech/common/covers/filter/OreDictionaryItemFilter.java

```diff
@@ -50,7 +50,7 @@ public void initUI(Consumer<Widget> widgetGroup) {
                 .setTooltip("cover.ore_dictionary_filter.info"));
         widgetGroup.accept(new ImageWidget(10, 25, 156, 14, GuiTextures.DISPLAY));
         widgetGroup.accept(new TextFieldWidget2(14, 29, 152, 12, () -> oreDictFilterExpression, this::setOreDictFilterExpression)
-                .setAllowedChars(Pattern.compile("[(!]* *[0-9a-zA-Z*]* *\\)*( *[&|^]? *[(!]* *[0-9a-zA-Z*]* *\\)*)*"))
+                .setAllowedChars(Pattern.compile("[0-9a-zA-Z* &|^!()]*"))
```

Comment:
Constructing regex patterns is still fairly expensive. This can be moved to a static constant to improve performance even further.

Reply:
Good point, thanks.